Applications/Calculator/CalculatorWidget.cpp

```diff
@@ -35,141 +37,105 @@
 
 CalculatorWidget::CalculatorWidget()
 {
-    set_fill_with_background_color(true);
 
-    m_entry = add<GUI::TextBox>();
+    load_from_gml(calculator_gml);
+    m_entry = *find_descendant_of_type_named<GUI::TextBox>("entry_textbox");
     m_entry->set_relative_rect(5, 5, 244, 26);
     m_entry->set_text_alignment(Gfx::TextAlignment::CenterRight);
     m_entry->set_font(Gfx::FontDatabase::default_fixed_width_font());
 
-    m_label = add<GUI::Label>();
-    m_label->set_relative_rect(12, 42, 27, 27);
+    m_label = *find_descendant_of_type_named<GUI::Label>("label");
+
     m_label->set_frame_shadow(Gfx::FrameShadow::Sunken);
     m_label->set_frame_shape(Gfx::FrameShape::Container);
     m_label->set_frame_thickness(2);
     auto label_palette = m_label->palette();
     label_palette.set_color(Gfx::ColorRole::WindowText, Color::Red);
     m_label->set_palette(label_palette);
```

Comment:
Seems like we can do this with the `foreground_color` property you added!

Reply:
done!